### ISSUE TITLE
For a single piece of data, the model performance test results are very weird.

### ISSUE THREAD

**Issue by Ziba-li** [airaria/TextPruner#9]
First of all, thank you for seeing my question.
I'm doing a bert-base four-category model tailoring: `target_ffn_size=1536, target_num_of_heads=6, head_even_masking=False, use_logits=True, pruning_method='iterative', n_iters=16`
The effect is as follows：
`100%|██████████| 14/14 [00:37<00:00,  2.69s/it]
cuda-warm-up: 100%|██████████| 5/5 [00:00<00:00, 106.38it/s]
cuda-repetitions: 100%|██████████| 10/10 [00:00<00:00, 133.73it/s]
Device: cuda:0
Mean inference time: 6.98ms
Standard deviation: 0.27ms
accuracy：82.84566838783705
100%|██████████| 14/14 [00:33<00:00,  2.36s/it]
cuda-warm-up: 100%|██████████| 5/5 [00:00<00:00, 135.14it/s]
cuda-repetitions: 100%|██████████| 10/10 [00:00<00:00, 137.01it/s]
Device: cuda:0
Mean inference time: 6.91ms
Standard deviation: 0.16ms
accuracy：74.69879518072288
100%|██████████| 14/14 [00:30<00:00,  2.17s/it]
cuda-warm-up: 100%|██████████| 5/5 [00:00<00:00, 416.77it/s]
cuda-repetitions: 100%|██████████| 10/10 [00:00<00:00, 416.65it/s]
Device: cuda:0
Mean inference time: 1.95ms
Standard deviation: 0.09ms
accuracy：82.09982788296041`
The performance test here is based on one data, but while the performance test results were almost identical to the original Bert-base ? (The final result is a hfl-rbt3 distillation model based on TextBrewer )
And thanks again.

**Comment by airaria** [comment]
There are several possibilities.
I encourage you to check the model size with `textpruner.summary` immediately after pruning.
And, have you saved your pruned model to the disk? 
Is the pruned model size smaller than BERT-base on the disk?
How did you test the speed? Did you test the speed immediately after pruning, or by loading the pruned model from the disk?

I would like to solve the issue If you could provide more details (by answering the above questions).

**Comment by airaria** [comment]
> First of all, thank you for seeing my question.
> I'm doing a bert-base four-category model tailoring: `target_ffn_size=1536, target_num_of_heads=6, head_even_masking=False, use_logits=True, pruning_method='iterative', n_iters=16`
> The effect is as follows：
> `100%|██████████| 14/14 [00:37<00:00, 2.69s/it] cuda-warm-up: 100%|██████████| 5/5 [00:00<00:00, 106.38it/s] cuda-repetitions: 100%|██████████| 10/10 [00:00<00:00, 133.73it/s] Device: cuda:0 Mean inference time: 6.98ms Standard deviation: 0.27ms accuracy：82.84566838783705 100%|██████████| 14/14 [00:33<00:00, 2.36s/it] cuda-warm-up: 100%|██████████| 5/5 [00:00<00:00, 135.14it/s] cuda-repetitions: 100%|██████████| 10/10 [00:00<00:00, 137.01it/s] Device: cuda:0 Mean inference time: 6.91ms Standard deviation: 0.16ms accuracy：74.69879518072288 100%|██████████| 14/14 [00:30<00:00, 2.17s/it] cuda-warm-up: 100%|██████████| 5/5 [00:00<00:00, 416.77it/s] cuda-repetitions: 100%|██████████| 10/10 [00:00<00:00, 416.65it/s] Device: cuda:0 Mean inference time: 1.95ms Standard deviation: 0.09ms accuracy：82.09982788296041`
> The performance test here is based on one data, but while the performance test results were almost identical to the original Bert-base ? (The final result is a hfl-rbt3 distillation model based on TextBrewer )
> And thanks again.

There are several possibilities.
I encourage you to check the model size with `textpruner.summary` immediately after pruning.
And, have you saved your pruned model to the disk? 
Is the pruned model size smaller than BERT-base on the disk?
How did you test the speed? Did you test the speed immediately after pruning, or by loading the pruned model from the disk?

I would like to solve the issue If you could provide more details (by answering the above questions).

**Comment by Ziba-li** [comment]
> > First of all, thank you for seeing my question.
> > I'm doing a bert-base four-category model tailoring: `target_ffn_size=1536, target_num_of_heads=6, head_even_masking=False, use_logits=True, pruning_method='iterative', n_iters=16`
> > The effect is as follows：
> > `100%|██████████| 14/14 [00:37<00:00, 2.69s/it] cuda-warm-up: 100%|██████████| 5/5 [00:00<00:00, 106.38it/s] cuda-repetitions: 100%|██████████| 10/10 [00:00<00:00, 133.73it/s] Device: cuda:0 Mean inference time: 6.98ms Standard deviation: 0.27ms accuracy：82.84566838783705 100%|██████████| 14/14 [00:33<00:00, 2.36s/it] cuda-warm-up: 100%|██████████| 5/5 [00:00<00:00, 135.14it/s] cuda-repetitions: 100%|██████████| 10/10 [00:00<00:00, 137.01it/s] Device: cuda:0 Mean inference time: 6.91ms Standard deviation: 0.16ms accuracy：74.69879518072288 100%|██████████| 14/14 [00:30<00:00, 2.17s/it] cuda-warm-up: 100%|██████████| 5/5 [00:00<00:00, 416.77it/s] cuda-repetitions: 100%|██████████| 10/10 [00:00<00:00, 416.65it/s] Device: cuda:0 Mean inference time: 1.95ms Standard deviation: 0.09ms accuracy：82.09982788296041`
> > The performance test here is based on one data, but while the performance test results were almost identical to the original Bert-base ? (The final result is a hfl-rbt3 distillation model based on TextBrewer )
> > And thanks again.
> 
> There are several possibilities. I encourage you to check the model size with `textpruner.summary` immediately after pruning. And, have you saved your pruned model to the disk? Is the pruned model size smaller than BERT-base on the disk? How did you test the speed? Did you test the speed immediately after pruning, or by loading the pruned model from the disk?
> 
> I would like to solve the issue If you could provide more details (by answering the above questions).
Thank you very mush.
The previous evaluation code was written inaccurately, and after a few tweaks the performance gap became apparent. And I have written according to `inference_time` onnx performance evaluation code, the effect is very good, this problem has been solved. Thanks again.

**Comment by Ziba-li** [comment]
> 

And if you have free time, you can take a look at the warehouse, I have been modified, it should not appear the above problems.
https://github.com/Ziba-li/template_classify/tree/main/optimize/evaluate
